extensions/amp-story/1.0/amp-story-system-layer.js

```diff
@@ -454,9 +457,12 @@ export class SystemLayer {
     if (!this.isBuilt_) {
       return;
     }
-    this.vsync_.mutate(() => {
-      this.getShadowRoot().setAttribute(MESSAGE_DISPLAY_CLASS, 'noshow');
-    });
+    this.currentClicks_--;
```

Comment:
Actually, `this.timer_.delay(...)` returns an ID that you can use to cancel the timer.  It may make more sense to just store that and cancel any previous timers each time.  GitHub doesn't let me comment on lines you didn't change, but in `hideAfterTimeout_()` you should be able to do:

```javascript
hideAfterTimeout_() {
  if (this.timeoutId_) {
    this.timer_.cancel(this.timeoutId_);
  }
  this.timeoutId_ = this.timer_.delay(() => this.hideInteral_(), hideTimeout);
}
```

Reply:
Ok cool :) Sorry for not exploring that before!